### PR TITLE
Make use of urlForDataItem

### PR DESCRIPTION
**Pull request by apepper**
Content needed: https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://urlfordataitem.scrivito-portal-app.pages.dev/my-tynacoon-59274260f223b65f?_scrivito_workspace_id=h0bfd49412972186&_scrivito_display_mode=view

### CODE REVIEW

File: src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx [hunk 1]
```diff
@@ -84,17 +86,12 @@ provideComponent(DataFormContainerWidget, ({ widget }) => {
...
     if (redirectAfterSubmit) {
-      // TODO: Remove this work around once #10212 is resolved
-      navigateTo(redirectAfterSubmit, {
-        params: {
-          [`${snakeCase(targetDataItem.dataClass().name())}_id`]:
-            targetDataItem.id(),
-        },
-      })
+      const url = await load(() => urlForDataItem(targetDataItem))
+      if (url) navigateTo(new Link({ url }))
```
Comment (dcsaszar) [inline src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:94]: * [ ] FYI The indirection through `new Link` looks like a workaround. We could use `if (url) getHistory()?.push(url)` instead.

**Review by dcsaszar — APPROVED** [review]
I don't have access to the mentioned content but tested it myself. LGTM. See comment for a potential refactoring.

**Comment by apepper** [comment]
FYI: I just granted you access to the working copy. Sorry for not adding you earlier.

File: src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx (hunk 1 above)
Comment (apepper) [reply src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:94]: This is a workaround for issue #10629 in the SDK.

Comment (dcsaszar) [reply src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:94]: 😆 So my impression was correct. Using history is better as it's not a workaround but a valid alternative implementation in the light of a missing convenience feature.

**Comment by dcsaszar** [comment]
> FYI: I just granted you access to the working copy. Sorry for not adding you earlier.

I think the toggle on https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://urlfordataitem.scrivito-portal-app.pages.dev/create-ticket-f909e9691db27360?_scrivito_workspace_id=h0bfd49412972186&_scrivito_display_mode=diff should be set to "Yes".

File: src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx (hunk 1 above)
Comment (apepper) [reply src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:94]: But history "destroys" the current app state, if I'm not mistaken. So all objs and data needs to be loaded again. With `navigateTo` it is more seamless.

Comment (dcsaszar) [reply src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:94]: > But history "destroys" the current app state

No, see https://github.com/Scrivito/scrivito-portal-app/blob/d696a2077b4c27610f2cce31fbf65f7ed33d199b/src/config/history.ts#L9

Comment (apepper) [reply src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx]: Good point!

**Comment by apepper** [comment]
REVIEW ([f4a3dc8](https://github.com/Scrivito/scrivito-portal-app/pull/249/commits/f4a3dc8a816d60a9d5571abd62c447a5b6eedb28))

**Pull request merged**